src/Redis/Orleans.Streaming.Redis/Streams/RedisStreamAdapterReceiver.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Serialization;
+using Orleans.Streams;
+
+namespace Orleans.Streaming.Redis;
+
+internal sealed class RedisStreamAdapterReceiver : IQueueAdapterReceiver
+{
+    private readonly ILogger _logger;
+    private readonly Serializer<RedisStreamBatchContainer> _serializer;
+    private RedisStreamStorage _queue;
+    private long _seqNumber;
```

Comment:
I think we should start from `DateTime.UtcNow.Ticks` and then increment the value, like we do for `MemoryStreamQueueGrain`. I know some provider starts always from zero, but it could cause issue during rolling upgrade.

Reply:
Actually, would it be possible to generate the `SequenceNumber` in Redis itself? SO we could make this provider rewindable.

Reply:
The message of Redis stream has its own id. So if we make each event generate a separate message, then it can be rewindable. But I'm concerned that this will drastically reduce throughput. In addition, I am not very familiar with rewindable related interfaces -_-